cvat/apps/engine/media_extractors.py

```diff
@@ -39,7 +39,7 @@ def __init__(self, source_path, dest_path, image_quality, step=1, start=0, stop=
         if not source_path:
             raise Exception('No image found')
         super().__init__(
-            source_path=sorted(source_path),
+            source_path=source_path,
```

Comment:
It is a very significant change. The primary problem that we can read the list of images from a directory (client directory, server share directory). OS doesn't guarantee the order. Thus the `sorted` operation is the way to make the process predictable. When you want to generate a pre-annotation you have to know the order of images. Using the technic (`sorted` operation) you can "repeat" the order locally in your script.

Thus the solution cannot be accepted. Need to find a more general one.

Reply:
Yes, OS doesn't guarantee the order of the directory contents. But for the directory case, I think it is already uniformed in [DirectoryExtractor](https://github.com/opencv/cvat/blob/develop/cvat/apps/engine/media_extractors.py#L135).

Reply:
But if you send a list of files from a browser (file dialog) you cannot control its order. You even cannot know the order. Right?

I believe the only way to solve the problem is to use filename (or path) as ID instead of indexes on the server. If you can prepare such patch we will accept it.

For now I will close the PR because it will give us some problems. I understand that from the command line you can specify the order of files. But from UI it isn't possible.